### PR TITLE
Schema convenience functions

### PR DESCRIPTION
**Pull request by byronellis**
While spending a bunch of times working with writing down schemas in Java recently one of the things that kept coming up was the need to make fields nullable more often than not. While there are convenience functions for NOT NULL fields there aren't equivalent nullable versions. This is just a small PR that adds equivalent convenience methods to the schema builder for nullable fields. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java [hunk 1]
```diff
@@ -142,82 +142,163 @@ public Builder addByteField(String name) {
       return this;
     }
 
+    public Builder addNullableByteField(String name) {
+      fields.add(Field.nullable(name, FieldType.BYTE));
```
Comment (chamikaramj) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java:146]: Probably we should use the existing "addNullableField" function when implementing these.

**Review by chamikaramj — COMMENTED** [review]
LGTM. Thanks.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java (hunk 1 above)
Comment (byronellis) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java:146]: sure, no problem

**Review by chamikaramj — APPROVED** [review]
Thanks!

**Comment by chamikaramj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by byronellis** [comment]
Yay, everything passed finally.

**Pull request merged**